pkg/querier/stats/stats.pb.go

```diff
@@ -54,19 +54,21 @@ type Stats struct {
 	SamplesProcessed uint64 `protobuf:"varint,11,opt,name=samples_processed,json=samplesProcessed,proto3" json:"samples_processed,omitempty"`
 	// The number of subqueries that were spun off as actual range queries in order to execute the full query
 	SpunOffSubqueries uint32 `protobuf:"varint,12,opt,name=spun_off_subqueries,json=spunOffSubqueries,proto3" json:"spun_off_subqueries,omitempty"`
+	// SamplesProcessedPerStep represents the number of samples scanned per step timestamp while evaluating a query.
+	SamplesProcessedPerStep []StepStat `protobuf:"bytes,13,rep,name=samples_processed_per_step,json=samplesProcessedPerStep,proto3" json:"samples_processed_per_step"`
```

Comment:
Have you considered having a different format for this? Right now if we wanted to add another type of statistics that we want to track per step, we would have to duplicate this array. So we'd have two arrays: one for SamplesProcessed and one for the next statistic. At least the timestamps would repeat.

WDYT about having single array where every element is a timestamp and a multiple values which have names. So in this case the field here would be called PerStepStatistics. And then the StepStats would be something like {Timestamp, SamplesProcessed}. (again - not sure if we need to store every timestamp or if start,end,step are enough)

Reply:
We had a chat about it. Passing PerStepStats is in fact adding just adding one more series to the response, so we shouldn't be too concerned about it.